### ISSUE TITLE
"Index out of bounds" when navigating external HTML

### ISSUE THREAD

**Issue by twogee** [asciidoctor/asciidoctor-intellij-plugin#335]
AsciiDoc Plugin Version: 0.30.6

IDE Name and Version: IntelliJ 2019.2.2

When following links to external sites in an asciidoc file, the external site is loaded in a preview window. However, there's no obvious way to navigate back. I tried randomly clicking some links and caused the following exception. I do not have a simple test case yet, but maybe the stack trace below could help. I triggered the exception several times with different index values (-12, -15, ...). 

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -15
	at java.base/java.lang.String.substring(String.java:1841)
	at org.asciidoc.intellij.editor.javafx.JavaFxHtmlPanel.saveImage(JavaFxHtmlPanel.java:324)
	at org.asciidoc.intellij.editor.javafx.JavaFxHtmlPanel.lambda$null$1(JavaFxHtmlPanel.java:311)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:312)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:433)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:416)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:878)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:466)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:704)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:465)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by ahus1** [comment]
Thanks for reporting this. This is a regression: when clicking on an external link in the preview, the link should open in the systems browser only. The preview should not change.

Rationale behind this: the preview based on JavaFX is usually not very good in showing external web sites. The system browser is more capable doing this, and allow full interaction with the site. For some of my projects I use an AsciiDoc file as a bookmark list that opens the browser on pages related to the project (test environments, etc.)

The issue you found is fixed in version 0.30.7 of the plugin that is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the IntelliJ plugin repository.

**Comment by twogee** [comment]
Works great, thanks!
